api/v1alpha1/stardoguser_types.go

```diff
@@ -25,19 +26,42 @@ import (
 
 // StardogUserSpec defines the desired state of StardogUser
 type StardogUserSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// StardogInstance describes a Stardog instance
+	StardogInstance string `json:"stardogInstance,omitempty"`
+	// StardogUserCredentialsSpec describes the credentials of a Stardog user
+	Credentials StardogUserCredentialsSpec `json:"credentials,omitempty"`
+	// Roles describe a list of roles assigned to a Stardog user
+	Roles []string `json:"roles,omitempty"`
+}
 
-	// Foo is an example field of StardogUser. Edit StardogUser_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+// StardogUserCredentialsSpec specifies the password of a Stardog user
+type StardogUserCredentialsSpec struct {
+	// Namespace describes where the user credentials are located in the cluster
+	// Defaults to metadata.namespace
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+	// SecretRef describes a reference to a secret where the credential is contained
+	SecretRef corev1.SecretKeySelector `json:"secretRef,omitempty"`
```

Comment:
A string and namespace combination is enough to specify a Secret.
```suggestion
	SecretRef string `json:"secretRef,omitempty"`
```